documentation/modules/exploit/multi/http/opmanager_sumpdu_deserialization.md

```diff
@@ -0,0 +1,116 @@
+12.5.328.
+
+#### CVE-2020-28653
+This vulnerability affects OpManager versions 12.1 - 12.5.232. The vulnerability involves sending a malicious PDU to the
+SmartUpdateManager handler that when deserialized executes an arbitary OS command.
+
+#### CVE-2021-3287
+This vulnerability is a patch bypass for CVE-2020-28653 and affects OpManger versions 12.5.233 - 12.5.328. When the
+original vulnerability was patched, it was done so using a new `ITOMObjectInputStream` deserializer class. This object
+has a flaw in it's validation logic. The object works by requiring the caller to specify a list of one or more object
+classes that can be deserialized. If an instance is used to perform more than one `readObject` call however, only the
+first is protected because once a serialized object of an allowed type is read from the stream, the
+`ITOMObjectInputStream` instance remains in a sort of authenticated state where subsequent objects can be read of any
+type.
+
+The exploit technique for this CVE leverages this by first sending a legitimate, serialized SUMPDU to create an instance
+of the `SUMServerIOAndDataAnalyzer` object whose `process` method makes multiple `readObject` calls using the same
+instance for each.
+
+Unlike exploiting CVE-2020-28653, to exploit CVE-2021-3287 the target server must have the SUM server running. This is
+not the case for the standard installer, but is the case for "Central" variant. Without the SUM server running, the log
+handler is not initialized which causes the request handler to crash making the vulnerable code path inaccessible.
+
+### Setup (Windows)
+
+1. Download an affected version for either Windows or Linux from the [archive][0]
+1. Run the installer executable
+1. Accept the default values for all settings (skip registration), until the very end when prompted to start the
+  application
+1. Unselect the option to start the application
+    1. If this option is missed, just navigate to the tray icon where it will say that it's starting and select the
+      option to stop it
+1. Start a command prompt as an administrative user
+1. Navigate to `C:\Program Files\ManageEngine\OpManager\bin`, older versions use `C:\ManageEngine\OpManager\bin`
+1. Run `run.bat`
+1. View and accept the license terms
+1. Press `f` to run the product in Free mode
+
+OpManager should start successfully after a few minutes. At that point the service can be exploited. In this case the
+session will be opened in the context of the user that ran the service with `run.bat`. Once the server is restarted and
+OpManager starts automatically, the vulnerability can be exploited to open a session in the context of NT
+AUTHORITY\SYSTEM.
+
+### Setup (Linux)
+
+1. Download an affected version for either Windows or Linux from the [archive][0]
+1. Run the installer executable as root
+1. Accept the default values for all settings (skip registration)
+1. Navigate to `/opt/ManageEngine/OpManagerCentral/bin`
```

Comment:
Looks like older Linux versions might be under the `/opt/ManageEngine/OpManager/bin` path, so that could be added here